db_sql.go

```diff
@@ -300,7 +302,7 @@ func (x *sqlUserStoreDB) GetIdentityFromUserId(userId UserId) (string, error) {
 }
 
 func (x *sqlUserStoreDB) GetUserIdFromIdentity(identity string) (UserId, error) {
-	row := x.db.QueryRow(`SELECT userid FROM authuserstore WHERE LOWER(email) = $1 AND (archived = false OR archived IS NULL)`, CanonicalizeIdentity(identity))
+	row := x.db.QueryRow(`SELECT userid FROM authuserstore WHERE LOWER(email) = $1 OR LOWER(username) = $1 AND (archived = false OR archived IS NULL)`, CanonicalizeIdentity(identity))
```

Comment:
Before committing this work, we need to document exactly what "identity" means... IIRC in our last discussion Identity was going to mean "Either a username or an email address. If ambiguous, the system must try both".